gensim/models/coherencemodel.py

```diff
@@ -91,7 +101,7 @@ def __init__(self, model=None, topics=None, texts=None, corpus=None, dictionary=
         else:
             self.dictionary = dictionary
```

Comment:
what is the point of checking `if isinstance(model.id2word, FakeDict):` above? why is it note enough to check for None?